### ISSUE TITLE
rngd takes more CPU when writing to /dev/urandom than /dev/random

### ISSUE THREAD

**Issue by c0D3M** [nhorman/rng-tools#156]
version: 6.7 , no hardware random number generator present on my system  
Since random get block when there isnt enough entropy, I moved to urandom.
This started to works but occassionally I see high CPU like 100 (25% in irix off mode)
When this happen I noticed following prints continuosly , and the same doesnt happen when output device is random.
Please suggest what could be causing this behaviour.

Much Thanks.

```
Added 4094/4096 bits entropy
Pool full at 4096, sleeping!
Added 4094/4096 bits entropy
Pool full at 4096, sleeping!
Added 4094/4096 bits entropy
Pool full at 4096, sleeping!
Added 4094/4096 bits entropy
Pool full at 4096, sleeping!
Added 4094/4096 bits entropy
Pool full at 4096, sleeping!
Added 4094/4096 bits entropy
Pool full at 4096, sleeping!
Added 4094/4096 bits entropy
Pool full at 4096, sleeping!
```

**Comment by nhorman** [comment]
dont set your random input device to urandom. urandom, by definition does not block, and so the write call to add entropy returns immediately and we try to add more.  high cpu usage is common in rngd when using jitterentropy and pulling lots of data from /dev/random.  I would suggest lowering the number of threads jitter uses to prevent cpu saturation.  Also suggest moving to the latest rngd (v6.15), there are several fixes to mitigate high cpu usage

**Comment by c0D3M** [comment]
Thanks Neil for quick reply.
If I dont set rngd output to urandom , other application blocks on random when entropy drops low.
I am using 2 thread for now, are you suggesting to go even lower ?  what rngd command switch does this as I dont find any in help.

```
Limiting thread count to 2 active cpus
JITTER starts 2 threads
JITTER thread on cpu 0 wakes up for refill
JITTER thread on cpu 1 wakes up for refill
```

**Comment by nhorman** [comment]
But if you do set rngd output to urandom, you're doing it wong.  urandom isn't set to allow for user space to add entropy to the kernel.  It doesn't matter if you're avoiding your initial problem or not.

The number of threads you are using should depend on the number of cpus you have.  Nominally theres one jitterentropy thread for every cpu in the system.  You can lower it with the -O jitter:num_threads:<v> option (see the man page)

**Comment by c0D3M** [comment]
So I reverted back to random , cpu usage back to normal but I see a problem sometime. System entropy goes and down rngd doesn't replenish it , it seems rngd is waiting on something, when i checked with strace attaching rngd pid, it shows waiting in read system call. 
What could be going wrong ? This happens sometimes only, Any other debug information  or you want me to try with some command line switch ? I tried -s -W but no luck.
